inginious/frontend/plugins/user_management/pages/api/user_data.py

```diff
@@ -0,0 +1,139 @@
+    hash_link = get_user_activation_link(user_final_info["username"], collection_manager)
+    post_data_auth_prob = _("- If at some point you had authentication problems, it may be due to the change process.")
+    post_data_contact_admin = _("- If you think any change is wrong, please contact the administrator.")
+    post_data_hash = _(
+        """- We have noticed that you have not yet activated your account, please click on the following link:
+        """) + hash_link
+    final_post = post_data_hash if hash_link else post_data_auth_prob
+    message = _("""Some changes have been made to your account:
+""") + get_changes() + _(""" To keep in mind:
+    """) + post_data_contact_admin + """ 
+    """ + final_post
+    try:
+        web.sendmail(web.config.smtp_sendername, user_email, subject, message)
+    except (ValueError, TypeError):
+        raise api.APIError(500, _("Something went  wrong when we sent the information email to the user"))
+
+
+def get_user_activation_link(username, collection_manager):
+    """ Returns the activation link if the user has not activated the account """
+    user = collection_manager.make_find_one_request("users", {"username": username})
+    if "activate" in user:
+        return web.ctx.home + "/register?activate=" + user["activate"]
+    return ""
+
+
+class UserDataAPI(SuperadminAPI):
+    """ API to get information about a user """
+
+    def API_GET(self):
+        """ Get request. Returns data about a user """
+        self.check_superadmin_rights()
+        username = get_mandatory_parameter(web.input(), "username")
+        user_data = self.get_user_data(username)
+        return 200, user_data
+
+    def API_POST(self):
+        """ request to change one or more of the basic data of a user: username, realname or email """
+        self.check_superadmin_rights()
+        flag = False
+        user_data = web.input()
+        username = get_mandatory_parameter(user_data, "username")
+        collections_manager = CollectionsManagerSingleton.get_instance()
+        user_original_info = self.get_user_data(username)
+
+        username_count = 0
+        email_count = 0
+        name_count = 0
+        try:
+            block_user(username, collections_manager)
+        except api.APIError as error:
+            return error.status_code, {"error": error.return_value}
+        if "email" in user_data:
+            flag = True
+            email_count = change_email(username, user_data["email"], collections_manager)
+        if "name" in user_data:
+            flag = True
+            name_count = change_name(username, user_data["name"], collections_manager)
+        if "new_username" in user_data:
+            flag = True
+            collection_name_list = json.loads(get_mandatory_parameter(user_data, "collection_list"))
+            username_count = change_username(username, user_data["new_username"], collections_manager,
+                                             collection_name_list)
```

Comment:
These values need to be checked here too, that is checking email format, etc. Remember that these APIs might be used from other clients different to the frontend.